app/util.py

```diff
@@ -11,8 +11,61 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import re
+from dateparser import DateDataParser
+from functools import lru_cache
+
+parser = DateDataParser(languages=['en'])
 
 def canonical_wikilink(wikilink):
+
+    if is_journal(wikilink):
+        try:
+            wikilink = canonical_date(wikilink)
+        except:
+            # TODO: if we add logging, maybe log that we couldn't parse a date here
+            pass
+
     # hack hack
-    wikilink = wikilink.lower().replace(' ', '-').replace('\'', '').replace(',', '')
+    wikilink = (
+        wikilink.lower()
+        .replace(' ', '-')
+        .replace('\'', '')
+        .replace(',', '')
+        .replace('/', '-')
+    )
     return wikilink
+
+
+@lru_cache(maxsize=None)
+def canonical_date(wikilink):
+    date = parser.get_date_data(wikilink).date_obj
+    try:
+        new_wikilink = date.isoformat().split("T")[0] 
+    except:
+        pass
+
+    return new_wikilink
+
+
+@lru_cache(maxsize=1)  #memoize this
+def get_combined_date_regex():
```

Comment:
I wonder if instead we could just call dateparser.parse(wikilink) and depend on its internal date inference logic, which might be more robust than a regex -- it returns None when it can't parse the string as a date, so in principle that could work.

Now, admittedly, for all we know it might be one or more of 1. too slow to call for each node to check if the node is a journal and 2. too fuzzy and mark some non-journal entries as a journal (e.g. a node called '2020').

Mmm, point 2. just convinced me a regex is better probably :) Sending this out for completeness, no action required.

Reply:
👌 sounds good. just fyi, the date parser does seem pretty slow. I added the lru_caches cause it noticeably impacted page loads, so probably best to keep it out of the loop unless it's needed.

Wouldn't be hard to remove it completely and just use regex match groups if performance becomes more important, but this was quicker for now